src/util/kafka/utility.js

```diff
@@ -0,0 +1,429 @@
+ *
+ * @param {string} template - The template for that needs to be populated
+ * @param {string} functionality - the functionality flow. Example: 'transfer'
+ * @param {string} action - the action that applies to the flow. Example: 'prepare'
+ *
+ * @returns {string} - Returns topic name to be created, throws error if failure occurs
+ */
+const transformGeneralTopicName = (template, functionality, action) => {
+  try {
+    if (Enum.Kafka.TopicMap[functionality] && Enum.Kafka.TopicMap[functionality][action]) {
+      return generalTopicTemplate(template, Enum.Kafka.TopicMap[functionality][action].functionality, Enum.Kafka.TopicMap[functionality][action].action)
+    }
+    return generalTopicTemplate(template, functionality, action)
+  } catch (err) {
+    Logger.error(err)
+    throw err
+  }
+}
+
+/**
+ * @function TransformGeneralTopicName
+ *
+ * @description participantTopicTemplate called which generates a participant topic name from the 3 inputs, which are used in the placeholder participant topic template found in the default.json
+ *
+ * @param {string} template - The template for that needs to be populated
+ * @param {string} participantName - participant name, retrieved from database. Example: 'dfsp1'
+ * @param {string} functionality - the functionality flow. Example: 'transfer'
+ * @param {string} action - the action that applies to the flow. Example: 'prepare'
+ *
+ * @returns {string} - Returns topic name to be created, throws error if failure occurs
+ */
+const transformAccountToTopicName = (template, participantName, functionality, action) => {
+  try {
+    return participantTopicTemplate(template, participantName, functionality, action)
+  } catch (err) {
+    Logger.error(err)
+    throw err
+  }
+}
+
+/**
+ * @function GetKafkaConfig
+ *
+ * @description participantTopicTemplate called which generates a participant topic name from the 3 inputs, which are used in the placeholder participant topic template found in the default.json
+ *
+ * @param {string} kafkaConfig - This is the Kafka config set in the default.json in you API
+ * @param {string} flow - This is required for the config for the Stream Processing API. Example: 'CONSUMER' ie: note the case of text
+ * @param {string} functionality - the functionality flow. Example: 'TRANSFER' ie: note the case of text
+ * @param {string} action - the action that applies to the flow. Example: 'PREPARE' ie: note the case of text
+ *
+ * @returns {string} - Returns topic name to be created, throws error if failure occurs
+ */
+const getKafkaConfig = (kafkaConfig, flow, functionality, action) => {
+  try {
+    const flowObject = kafkaConfig[flow]
+    const functionalityObject = flowObject[functionality]
+    const actionObject = functionalityObject[action]
+    actionObject.config.logger = Logger
+    return actionObject.config
+  } catch (e) {
+    throw new Error(`No config found for flow='${flow}', functionality='${functionality}', action='${action}'`)
+  }
+}
+
+/**
```

Comment:
Please move this outside of Kafka since it should be message transport agnostic.

E.g. `/src/util/streaming/protocol/index.js`